lightning/src/ln/channel.rs

```diff
@@ -850,8 +849,8 @@ impl<Signer: Sign> Channel<Signer> {
 		let holder_signer = keys_provider.get_channel_signer(false, channel_value_satoshis);
 		let pubkeys = holder_signer.pubkeys().clone();
 
-		if channel_value_satoshis >= MAX_FUNDING_SATOSHIS {
-			return Err(APIError::APIMisuseError{err: format!("funding_value must be smaller than {}, it was {}", MAX_FUNDING_SATOSHIS, channel_value_satoshis)});
+		if channel_value_satoshis >= TOTAL_BITCOIN_SUPPLY_SATOSHIS {
+			return Err(APIError::APIMisuseError { err: format!("funding_value must be smaller than the total bitcoin supply, it was {}", channel_value_satoshis) });
```

Comment:
Should we also return an error if our counterparty doesn't support wumbo but we try to open a channel >= 2^24?

Reply:
I don't see why we should? Its extra code that doesn't protect against anything. Not sure what it'd accomplish.

Reply:
Mainly because it's a MUST in the spec:

```
The sending node:
- if both nodes advertised option_support_large_channel:
  - MAY set funding_satoshis greater than or equal to 2^24 satoshi.
- otherwise:
  - MUST set funding_satoshis to less than 2^24 satoshi.
```

Reply:
Isn't this code the sending side for `open_channel`? `ChannelManager::create_channel` call `Channel::new_outbound` prior to generating `events::MessageSendEvent::SendOpenChannel`.

Reply:
Oh! I'm an idiot! Yes, we should have a check on the sending side, IMO.